receiver/swohostmetricsreceiver/internal/scraper/framework/scope/emitter.go

```diff
@@ -133,32 +134,23 @@ func processMetricEmitter(
 	rCh chan *metricEmitterEmitResult,
 	meName string,
 	me metric.Emitter,
+	logger *zap.Logger,
 ) {
 	defer emitWg.Done()
 
-	zap.L().Sugar().Debugf(
-		"emitting of metric emitter for metric '%s'",
-		meName)
-
+	logger.Debug("emitting of metric emitter", zap.String("metric", meName))
 	mr := me.Emit()
 	rCh <- &metricEmitterEmitResult{mr, meName}
 }
 
-func processEmittedResults(
-	rCh chan *metricEmitterEmitResult,
-) ([]pmetric.MetricSlice, error) {
+func processEmittedResults(rCh chan *metricEmitterEmitResult) ([]pmetric.MetricSlice, error) {
 	errs := make([]error, 0)
 	mrs := make([]pmetric.MetricSlice, 0)
 
 	// Collect results.
 	for r := range rCh {
 		if r.Result.Error != nil {
-			message := fmt.Sprintf(
-				"emit for metric emitter '%s' failed",
-				r.EmitterName,
-			)
-			zap.L().Error(message, zap.Error(r.Result.Error))
-			errs = append(errs, fmt.Errorf("%s: %w", message, r.Result.Error))
+			errs = append(errs, fmt.Errorf("emit for metric emitter '%s' failed: %w", r.EmitterName, r.Result.Error))
```

Comment:
`errrs` should be just `err`
```suggestion
			Errors.join(err, fmt.Errorf("emit for metric emitter '%s' failed: %w", r.EmitterName, r.Result.Error))
```
and imho join should be used. But feel free to skip!!! 🙈